File: VB/WindowsApplication53/CustomButtonEdit.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo
Imports System.ComponentModel
Imports DevExpress.XtraPrinting

Namespace WindowsApplication53

    'The attribute that points to the registration method
    <UserRepositoryItem("RegisterCustomButtonEdit")>
    Public Class RepositoryItemCustomButtonEdit
        Inherits RepositoryItemButtonEdit

        'The static constructor which calls the registration method
        Shared Sub New()
            Call RegisterCustomEdit()
        End Sub

        'The unique name for the custom editor
        Public Const CustomEditName As String = "CustomButtonEdit"

        'Return the unique name
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return CustomEditName
            End Get
        End Property

        'Register the editor
        Public Shared Sub RegisterCustomEdit()
            Call EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(CustomEditName, GetType(CustomButtonEdit), GetType(RepositoryItemCustomButtonEdit), GetType(ButtonEditViewInfo), New ButtonEditPainter(), True))
        End Sub

        'Return custom brick to print image within the cells
        Public Overrides Function GetBrick(ByVal info As PrintCellHelperInfo) As IVisualBrick
            Dim baseBrick As TextBrick = TryCast(MyBase.GetBrick(info), TextBrick)
            If baseBrick Is Nothing Then Return MyBase.GetBrick(info)
            Dim panelBrick As IPanelBrick = info.PS.CreatePanelBrick()
            Dim imageBrick As IImageBrick = info.PS.CreateImageBrick()
            panelBrick.Bricks.Add(baseBrick)
            panelBrick.Bricks.Add(imageBrick)
            imageBrick.Padding = New PaddingInfo(2, 2, 5, 5)
            imageBrick.Image = Buttons(0).Image
            imageBrick.Sides = BorderSide.None
            baseBrick.Sides = BorderSide.None
            baseBrick.Rect = New System.Drawing.RectangleF(info.Rectangle.Height - 6, 0, info.Rectangle.Width - info.Rectangle.Height + 6, info.Rectangle.Height)
            imageBrick.Rect = New System.Drawing.RectangleF(0, 0, info.Rectangle.Height - 6, info.Rectangle.Height)
            Return panelBrick
        End Function
    End Class

    Public Class CustomButtonEdit
        Inherits ButtonEdit

        'The static constructor which calls the registration method
        Shared Sub New()
            RepositoryItemCustomButtonEdit.RegisterCustomEdit()
        End Sub

        'Initialize the new instance
        Public Sub New()
        End Sub

        'Return the unique name
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemCustomButtonEdit.CustomEditName
            End Get
        End Property

        'Override the Properties property
        'Simply type-cast the object to the custom repository item type
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemCustomButtonEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomButtonEdit)
            End Get
        End Property
    End Class
End Namespace

File: CS/WindowsApplication53/CustomButtonEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;
using System.ComponentModel;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.NativeBricks;

namespace WindowsApplication53
{
    //The attribute that points to the registration method
    [UserRepositoryItem("RegisterCustomButtonEdit")]
    public class RepositoryItemCustomButtonEdit : RepositoryItemButtonEdit
    {
        //The static constructor which calls the registration method
        static RepositoryItemCustomButtonEdit() { RegisterCustomEdit(); }
        //The unique name for the custom editor
        public const string CustomEditName = "CustomButtonEdit";
        //Return the unique name
        public override string EditorTypeName { get { return CustomEditName; } }
        //Register the editor
        public static void RegisterCustomEdit()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(CustomEditName,
              typeof(CustomButtonEdit), typeof(RepositoryItemCustomButtonEdit),
              typeof(ButtonEditViewInfo), new ButtonEditPainter(), true));
        }
        //Return custom brick to print image within the cells
        public override VisualBrick GetBrick(PrintCellHelperInfo info)
        {
            TextBrick baseBrick = base.GetBrick(info) as TextBrick;
            if (baseBrick == null) return base.GetBrick(info);
            PanelBrick panelBrick = info.PS.CreateBrick("PanelBrick") as PanelBrick;
            ImageBrick imageBrick = info.PS.CreateBrick("ImageBrick") as ImageBrick;
            panelBrick.Bricks.Add(baseBrick);
            panelBrick.Bricks.Add(imageBrick);
            imageBrick.Padding = new PaddingInfo(2, 2, 5, 5);                        
            imageBrick.Image = this.Buttons[0].Image;
            imageBrick.Sides = BorderSide.None ;
            baseBrick.Sides = BorderSide.None;
            baseBrick.Rect = new System.Drawing.RectangleF(info.Rectangle.Height-6, 0, info.Rectangle.Width - info.Rectangle.Height + 6, info.Rectangle.Height);
            imageBrick.Rect = new System.Drawing.RectangleF(0, 0, info.Rectangle.Height - 6, info.Rectangle.Height);
            return panelBrick;
        }
    }


    public class CustomButtonEdit : ButtonEdit
    {
        //The static constructor which calls the registration method
        static CustomButtonEdit() { RepositoryItemCustomButtonEdit.RegisterCustomEdit(); }
        //Initialize the new instance
        public CustomButtonEdit() { }
        //Return the unique name
        public override string EditorTypeName
        {
            get
            {
                return   RepositoryItemCustomButtonEdit.CustomEditName;
            }
        }
        //Override the Properties property
        //Simply type-cast the object to the custom repository item type
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomButtonEdit Properties
        {
            get { return base.Properties as RepositoryItemCustomButtonEdit; }
        }
        

    }

}
